.github/workflows/build_rpm.yaml

```diff
@@ -0,0 +1,174 @@
+          echo 'export PATH=$PATH:/opt/rustup:/opt/cargo/bin' >> $GITHUB_PATH
+          packages=("unzip" "rpm-build" "openssl-devel" "curl" "gcc")
+          for package in "${packages[@]}"; do
+            if ! yum list installed "$package" >/dev/null 2>&1; then
+              yum install -y --allowerasing "$package"
+            fi
+          done
+          rustup default stable
+
+      - name: Setup for alumet
+        run: |
+          mkdir -p alumet
+          mkdir -p /etc/alumet  // TODO Change on merge !!
+
+      - name: Checkout alumet repo
+        uses: actions/checkout@v4
+        with:
+          # repository: alumet-dev/alumet
+          repository: AngeCyp/alumet-testing  # TODO Change on merge
+          path: "alumet-agent"
+          ref: '${{ inputs.tag }}'
+
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+          cache-targets: "alumet"
+
+      - name: get Spec settings
+        id: get-spec-settings
+        env:
+          build_dir: "$(pwd)/build"
+        run: |
+          mkdir -p ${{ env.build_dir }}/SOURCES
+          tar --exclude=".[^/]*" --exclude="./build" --transform 's|.|alumet-agent|' -zcvf ${{ env.build_dir }}/SOURCES/alumet-agent.tar.gz .
+          rpmbuild -ba -vv --define "_topdir ${{ env.build_dir }}" --define "release ${{ inputs.release-version }}" --define "osr ${{ env.OS }}" --define "version ${{ inputs.build-version }}" --define "arch ${{ inputs.target-architecture }}" packaging/rpm/specs/alumet.spec
+
+      - name: Upload artifact alumet-agent
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.name }}-agent-${{ inputs.build-version }}-${{ inputs.release-version }}.${{ env.OS }}.${{ inputs.target-architecture }}.rpm
+          path: |
+            ./build/RPMS/${{ inputs.target-architecture }}/${{ env.name }}-agent-${{ inputs.build-version }}-${{ inputs.release-version }}.${{ env.OS }}.${{ inputs.target-architecture }}.rpm
+
+  test-rpm:
+    runs-on: ubuntu-latest
+    needs:
+      - packaging
+    container:
+      image: "registry.access.redhat.com/ubi8/ubi:8.3"
+      options: --cap-add=SYS_NICE --cap-add=PERFMON
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          repository: 'alumet-dev/packaging'
+          fetch-depth: 1
+          path: 'packaging'
+          ref: 'ci/create-rpm-package'  # TODO Change on merge !!
+
+      - name: Download RPM artifact
+        id: download_rpm
+        uses: actions/download-artifact@v4
+        with:
+          pattern: '**/*8.3*.rpm'
+          path: ./rpm
+          merge-multiple: true
+
+      - name: Install RPMs
+        run: |
+          rpm -i ${{ steps.download_rpm.outputs.download-path }}/*.rpm
+
+      - name: Install Goss
+        run: |
+          curl -fsSL https://goss.rocks/install | sh
+
+      - name: Run Goss tests
```

Comment:
Maybe use a matrix in test workflow to test every RPM cause it sounds like installing all RPMs in step `Install RPMs` will override from one to the others and only the latest will be tested here. Also it will make sure RPMs works in the OS they've been built for, here everything run on ubi8.3 image)